src/components/_General/TokenMediaDisplay.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useState, useMemo } from 'react';
+import { ipcRenderer } from 'electron';
+import styled from '@emotion/styled';
+
+import { Responsive, getContentType } from 'util/helpers';
+import { V } from 'util/theming';
+import { IPFS_IPC_ID, IpfsAction, checkIsIPFSLink, mediaTypes, HTTP_ERR_405 } from 'vars/defines';
+
+import Loader from 'components/_General/Spinner';
+
+const MediaContent = styled.div`
+  overflow-y: auto;
+  display: flex;
+  width: 100%;
+  padding-left: 20px;
+  justify-content: center;
+  ${Responsive.below.L} {
+    order: 1;
+    padding-left: 0;
+  }
+`;
+
+const ImageFrame = styled.div`
+  border-radius: ${V.size.borderRadius};
+`;
+
+const TokenImage = styled.img`
+  max-width: 100%;
+`;
+
+const LoaderContainer = styled.div`
+  width: 200px;
+  height: 200px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  border: 1px solid ${V.color.backSoftest};
+`;
+
+interface TokenMediaDisplayProps {
+  url?: string;
+}
+
+const TokenMediaDisplay: React.FC<TokenMediaDisplayProps> = ({ url }) => {
+  const [contentType, setContentType] = useState<null | string | string[]>(null);
+  const [imageUrl, setImageUrl] = useState(null);
+
+  const isMedia = useMemo(
+    () => !!contentType && mediaTypes.includes(contentType as typeof mediaTypes[number]),
+    [contentType]
+  );
+
+  useEffect(() => {
```

Comment:
These two `useEffect`s should probably be combined into one, since their dependency array is identical.